### PR TITLE
Generalize components to support correction range overrides

### PR DESCRIPTION
**Pull request by mpangburn**
- Introduce `ExpandableSetting` to generalize over the existing card expansion used in `SuspendThresholdEditor` and `ScheduleItemView`
- Allow `GlucoseRangePicker` to be used either as a picker component or an independent picker
- Support titled sections in `CardList` (will be used for Delivery Limits page in the near future)

### CODE REVIEW

File: LoopKitUI/Views/ExpandableSetting.swift [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+public struct ExpandableSetting<
+    LeadingValueContent: View,
+    TrailingValueContent: View,
+    ValuePicker: View
+>: View {
+    @Binding var isEditing: Bool
+    var leadingValueContent: LeadingValueContent
+    var trailingValueContent: TrailingValueContent
+    var valuePicker: ValuePicker
```
Comment (nhamming) [inline LoopKitUI/Views/ExpandableSetting.swift:20]: [nit] Is this a true `Picker` or just has a similar name? If it is intended to be a true `Picker` all is good. If this is intended to allow other controls for input, consider renaming.

For true `Picker` scenario, is there a way to restrict the `ValuePicker` `View` type to that of a `Picker`?

**Review by nhamming — APPROVED** [review]
LGTM

**Review by ps2 — APPROVED** [review]
LGTM!

File: LoopKitUI/Views/ExpandableSetting.swift (hunk 1 above)
Comment (mpangburn) [reply LoopKitUI/Views/ExpandableSetting.swift:20]: Good idea; will rename to `ExpandedContent` in a follow-up PR

**Pull request merged**